File: frontend/src/app/models/users/user-info.ts
import { Image } from '../general/image';
import { SiteForm } from '../sites/site-form';
import { RoleForm } from '../roles/role-form';

export class UserInfo {
    public uuid: string;
    public firstName: string;
    public lastName: string;
    public email: string;
    public position: string;
    public contactNumber: string;
    public imageUuid: string;
    public password: string;
    public verifyPassword: string;
    public expireDate: string;
    public staffNumber: string;
    public imageInfos: Image[];
    public imageInfo: Image;
    public siteInfos: SiteForm[];
    public roleInfos: RoleForm[];

}

File: backend/src/main/java/com/minetec/backend/dto/info/UserInfo.java
package com.minetec.backend.dto.info;

import lombok.Data;

import javax.validation.constraints.NotNull;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;


@Data
public class UserInfo {

    @NotNull
    private String email;
    private UUID uuid;
    private String firstName;
    private String lastName;
    private String position;
    private String contactNumber;
    private LocalDateTime expireDate;
    private boolean locked;
    private String token;
    private List<RoleInfo> roleInfos;
    private List<String> permissions;
    private ImageInfo imageInfo;
    private List<SiteInfo> siteInfos;
    /**
     * TODO : current site is not ready
     */
    private SiteInfo currentSite;
    private String staffNumber;
}
